openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -0,0 +1,164 @@
+"""
+Provider:   iNaturalist
+
+Output:     TSV file containing the media metadata.
+
+Notes:      [The iNaturalist API is not intended for data scraping.]
+            (https://api.inaturalist.org/v1/docs/)
+
+            [But there is a full dump intended for sharing on S3.]
+            (https://github.com/inaturalist/inaturalist-open-data/tree/documentation/Metadata)
+
+            Because these are very large normalized tables, as opposed to more document
+            oriented API responses, we found that bringing the data into postgres first
+            was the most effective approach. [More detail in slack here.]
+            (https://wordpress.slack.com/archives/C02012JB00N/p1653145643080479?thread_ts=1653082292.714469&cid=C02012JB00N)
+
+            We use the table structure defined [here,]
+            (https://github.com/inaturalist/inaturalist-open-data/blob/main/Metadata/structure.sql)
+            except for adding ancestry tags to the taxa table.
+"""
+
+import os
+from pathlib import Path
+from typing import Dict
+
+import pendulum
+from airflow.exceptions import AirflowSkipException
+from airflow.operators.python import PythonOperator
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from airflow.providers.postgres.operators.postgres import PostgresOperator
+from airflow.utils.task_group import TaskGroup
+from common.constants import POSTGRES_CONN_ID
+from common.licenses import LicenseInfo, get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+AWS_CONN_ID = os.getenv("AWS_CONN_ID", "test_conn_id")
+PROVIDER = prov.INATURALIST_DEFAULT_PROVIDER
+SCRIPT_DIR = Path(__file__).parents[1] / "provider_csv_load_scripts/inaturalist"
+SOURCE_FILE_NAMES = ["photos", "observations", "taxa", "observers"]
+
+
+class INaturalistDataIngester(ProviderDataIngester):
+
+    providers = {"image": prov.INATURALIST_DEFAULT_PROVIDER}
+
+    def __init__(self, *kwargs):
+        super(INaturalistDataIngester, self).__init__()
+        self.pg = PostgresHook(POSTGRES_CONN_ID)
+
+        # adjustments to buffer limits. TO DO: try to integrate this with the dev
+        # environment logic in the base class, rather than just over-writing it.
+        self.media_stores["image"].buffer_length = 10_000
+        self.batch_limit = 10_000
```

Comment:
Out of curiosity: how was 10k decided on?

Reply:
Gut instinct from @AetherUnbound, if memory serves. I haven't been able to run the whole thing on my local machine, because of thermal throttling, but I did get through several hundred thousand (maybe a couple of million?) records locally.

Reply:
Gotcha. Is there a plan to monitor the resource consumption of this DAG when we first run it in production @AetherUnbound?

Reply:
That's right, our normal limit is 100 but since there's so much data here it seemed prudent to increase the batch volume on both ends. We will indeed be monitoring performance, and we'll be sure to kick this off for the first time when nothing else intensive is running on Airflow so we have the smallest likelyhood of taking something down (CC @stacimc)